doc/fundamentals/s3gateway.md

```diff
@@ -0,0 +1,178 @@
+# S3Gateway
+
+Pachyderm includes s3gateway, which allows you to interact with PFS storage
+through an HTTP API that imitates a subset of AWS S3's API. With this, you
+can reuse a number of tools and libraries built to work with object stores
+(e.g. [minio](https://docs.minio.io/docs/minio-client-quickstart-guide.html),
+[Boto](https://github.com/boto/boto3)) to interact with pachyderm.
+
+## Connecting to the s3gateway
+
+You can start the s3gateway locally by running `pachctl s3gateway`. Then point
+your browser or favorite s3 tool at `http://localhost:30600`. The s3gateway
+also runs in the background on your pachyderm cluster, which you can reach via
+`http://<cluster ip>:30600`.
+
+**Note:** A third way to access the s3gateway is using the explicit port
+forwarder (`pachctl port-forward`). However, we don't recommend it, as
+kubernetes' port forwarder incurs overhead, and does not recover well from
+broken connections.
+
+## Overview
```

Comment:
I think this section should be approached more as API docs rather than a full reference; the reason being is that users shouldn't be calling into this endpoints directly, but rather using tools/libraries designed to work with object stores, and abstract most things away. I added a caveat regarding that in the docs. I realize that makes this harder to review, though, if you haven't used S3 libraries in the past.

With regard to adding details on buckets, objects, and other S3- and HTTP-specific terminology, I'm not sure how to do so without really muddying the waters for users who already understand those concepts.

It's also difficult to give an exact mapping of endpoints to the equivalent `pachctl` commands, because there's a lot conditional behavior going on. I did add more details about what the various endpoints are doing though.

Let me know what you think!

Reply:
Ok. I'm fine if we treat this as more API docs and my teach then creates other docs that are more introductory

Reply:
Since we'll do it in that regard, can we move this from the `fundamentals` section and move it to the `reference` section of the docs? Docs team will reorganize that again if needed.